src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1776,8 +1755,25 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_Vector512_GreaterThanAll:
+        case NI_Vector512_GreaterThanOrEqualAll:
+        case NI_Vector512_LessThanAll:
+        case NI_Vector512_LessThanOrEqualAll:
```

Comment:
Why not pair these up with the corresponding `NI_Vector128/256` paths?

Reply:
These function differently from Vector256/128. I added a new function `LowerHWIntrinsicCmpOpWithKReg()` during lowering because there was essentially no common code from `LowerHWIntrinsicCmpOp()`. 

`NI_Vector128/256_GreaterThanAll()` ,` LesserThaAll() `etc uses` gtNewSimdCmpOpAllNode()` which basically reduces everything to `gtNewSimdCmpOpNode()` + `NI_Vector256_op_Equality` with `AllBitsSet`  since there are no separate instrinsics for GE, GT, LE, LT. With Vector512, we can simply use https://www.felixcloutier.com/x86/vpcmpd:vpcmpud with the correct `imm `and get the result without the extra  `NI_Vector256_op_Equality `with `AllBitsSet `part. It can be just `VPCMP* + KORTEST*`

Reply:
That sounds good for lowering. We do, however, want to try to keep the node creation for these "core" operations as part of the `gtNewSimd*Node` helpers  as this ensures that other phases of the JIT can easily say they want `GT_LT` for `simdSize: 16/32/64` for example, and have things "just work".

So it would be good to fix importation here to still go through `gtNewSimdCmpOpNode`; otherwise we're stuck duplicating the necessary size checks, any validation, and general switching logic whenever such an optimization comes up.

Reply:
Changed to use `gtNewSimdCmpAllOpNode`